src/configuration-gui/abrt-config-widget.c

```diff
@@ -120,10 +120,10 @@ static void abrt_config_widget_finalize(GObject *object);
 static AbrtAppConfiguration *
 abrt_app_configuration_new(const char *app_name)
 {
-    AbrtAppConfiguration *conf = libreport_xmalloc(sizeof(*conf));
+    AbrtAppConfiguration *conf = g_new(AbrtAppConfiguration, 1);
 
-    conf->app_name = libreport_xstrdup(app_name);
-    conf->settings = libreport_new_map_string();
+    conf->app_name = g_strdup(app_name);
```

Comment:
Is it okay to replace `libreport_xstrdup` with `g_strdup`? The former aborts with a message if there's not enough memory, the latter only returns `NULL`. I think we should at least check the return value if we're going to use the GLib functions here. (Same with most of the `libreport_x*` functions.)

Reply:
It is, GLib memory-allocation functions also abort.

Reply:
There’s even a recent commit to clarify this: https://gitlab.gnome.org/GNOME/glib/-/commit/445aa65e900cc2c0a861739b660f00d0b748a6f6.

Reply:
Ah, nice, didn't know that. I tried to follow the path down to the `_GLIB_NEW` macro but didn't find any mention of aborting.

Reply:
For science:
https://gitlab.gnome.org/GNOME/glib/-/blob/c93318953eec6a65de3e466f648a4f26109ff0eb/glib/gmem.h#L283 translates to `g_malloc()` being called, which has https://gitlab.gnome.org/GNOME/glib/-/blob/c93318953eec6a65de3e466f648a4f26109ff0eb/glib/gmem.c#L111 and which eventually ends up at https://gitlab.gnome.org/GNOME/glib/-/blob/c93318953eec6a65de3e466f648a4f26109ff0eb/glib/gmessages.c#L556.